ledger/ledger-runner-common/src/main/scala/com/daml/ledger/runner/common/PureConfigReaderWriter.scala

```diff
@@ -0,0 +1,319 @@
+    deriveReader[IndexerStartupMode.MigrateAndStart]
+  implicit val migrateAndStartWriter: ConfigWriter[IndexerStartupMode.MigrateAndStart] =
+    deriveWriter[IndexerStartupMode.MigrateAndStart]
+
+  implicit val validateAndWaitOnlyReader: ConfigReader[IndexerStartupMode.ValidateAndWaitOnly] =
+    deriveReader[IndexerStartupMode.ValidateAndWaitOnly]
+  implicit val validateAndWaitOnlyWriter: ConfigWriter[IndexerStartupMode.ValidateAndWaitOnly] =
+    deriveWriter[IndexerStartupMode.ValidateAndWaitOnly]
+
+  implicit val indexerStartupModeReader: ConfigReader[IndexerStartupMode] =
+    deriveReader[IndexerStartupMode]
+  implicit val indexerStartupModeWriter: ConfigWriter[IndexerStartupMode] =
+    deriveWriter[IndexerStartupMode]
+
+  implicit val haConfigReader: ConfigReader[HaConfig] = deriveReader[HaConfig]
+  implicit val haConfigWriter: ConfigWriter[HaConfig] = deriveWriter[HaConfig]
+
+  private def createParticipantId(participantId: String) =
+    Ref.ParticipantId
+      .fromString(participantId)
+      .left
+      .map(err => CannotConvert(participantId, Ref.ParticipantId.getClass.getName, err))
+
+  implicit val participantIdReader: ConfigReader[Ref.ParticipantId] = ConfigReader
+    .fromString[Ref.ParticipantId](createParticipantId)
+
+  implicit val participantIdWriter: ConfigWriter[Ref.ParticipantId] =
+    ConfigWriter.toString[Ref.ParticipantId](_.toString)
+
+  implicit val indexerConfigReader: ConfigReader[IndexerConfig] = deriveReader[IndexerConfig]
+  implicit val indexerConfigWriter: ConfigWriter[IndexerConfig] = deriveWriter[IndexerConfig]
+
+  implicit val sizedCacheReader: ConfigReader[SizedCache.Configuration] =
+    deriveReader[SizedCache.Configuration]
+  implicit val sizedCacheWriter: ConfigWriter[SizedCache.Configuration] =
+    deriveWriter[SizedCache.Configuration]
+
+  implicit val lfValueTranslationCacheReader: ConfigReader[LfValueTranslationCache.Config] =
+    deriveReader[LfValueTranslationCache.Config]
+  implicit val lfValueTranslationCacheWriter: ConfigWriter[LfValueTranslationCache.Config] =
+    deriveWriter[LfValueTranslationCache.Config]
+
+  implicit val indexConfigurationReader: ConfigReader[IndexServiceConfig] =
+    deriveReader[IndexServiceConfig]
+  implicit val indexConfigurationWriter: ConfigWriter[IndexServiceConfig] =
+    deriveWriter[IndexServiceConfig]
+
+  implicit val participantConfigReader: ConfigReader[ParticipantConfig] =
+    deriveReader[ParticipantConfig]
+  implicit val participantConfigWriter: ConfigWriter[ParticipantConfig] =
+    deriveWriter[ParticipantConfig]
+
+  implicit def participantNameKeyReader: ConfigReader[Map[ParticipantName, ParticipantConfig]] =
+    genericMapReader[ParticipantName, ParticipantConfig]((s: String) => Right(ParticipantName(s)))
+  implicit def participantNameKeyWriter: ConfigWriter[Map[ParticipantName, ParticipantConfig]] =
+    genericMapWriter[ParticipantName, ParticipantConfig](_.value)
```

Comment:
Is there any reason why these are defs?

Reply:
there was a reason but it disappeared after some further changes, thanks for pointing it out